tests/test_enums.py

```diff
@@ -219,3 +225,231 @@ def test_int_enum_arg_is_transformed_to_internal_value():
+    input_schema = """
+        type Query {
+            complex(i: Test = { role: EMPIRE }): String
+        }
+        input Test {
+            ignore: String 
+            role: Episode = EMPIRE  
+        }
+        input BetterTest {
+            newIgnore: String
+            test: Test = { role: ANDRZEJU }
+        }
+    """
+
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "Value for type: <Test> is invalid. "
+            "Check InputField/Arguments for <test> in <BetterTest> (Undefined enum value)."
+        ),
+    ):
+        make_executable_schema([enum_definition, input_schema])
+
+
+def test_input_exc_schema_should_raise_an_exception_if_undefined_enum_in_query():
+    input_schema = """
+        type Query {
+            complex(i: BetterTest = { test: { role: TWO_TOWERS } }): String
+        }
+        input Test {
+            ignore: String 
+            role: Episode = EMPIRE  
+        }
+        input BetterTest {
+            newIgnore: String
+            test: Test = { role: NEW_HOPE }
+        }
+    """
+
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "Value for type: <BetterTest> is invalid. "
+            "Check InputField/Arguments for <i> in <Query> (Undefined enum value)."
+        ),
+    ):
+        make_executable_schema([enum_definition, input_schema])
+
+
+def test_find_enum_values_in_schema_for_undefined_and_invalid_values():
+    input_schema = """
+        type Query {
+            complex(hello: Episode = EMPIRE,
+                    hi: Test = { role: JEDI,  next_role: ENTERPRISE, ignore: "HI"}, 
+                    bonjour: BetterTest = {newIgnore: "Witam", test: { role: NEWHOPE }}): String
+        }
+    
+        input Test {
+            ignore: String 
+            role: Episode = EMPIRE  
+            next_role: Episode
+        }
+
+        input BetterTest {
+            newIgnore: String
+            test: Test = { role: NEWHOPE }
+        }
+    """
+    query_complex_keys = [["next_role"], ["role"], ["test", "role"]]
+    better_test_complex_keys = [["role"]]
+    number_of_defined_enum_values = 7
+
+    # 2 Undefined because of "ENTERPRISE" invalid value, and next_role in Test input
+    number_of_undefined_default_enum_values = 3
+
+    ast_document = parse(join_type_defs([enum_definition, input_schema]))
+    schema = build_ast_schema(ast_document)
+    enums_entities = list(find_enum_values_in_schema(schema))
+    keys_to_complex_inputs = [keys for *_, keys in enums_entities if keys is not None]
+
+    undefined = [
+        (*_, args, keys)
+        for *_, args, keys in enums_entities
+        if args.default_value is Undefined
+    ]
+
+    assert keys_to_complex_inputs == query_complex_keys + better_test_complex_keys
+    assert len(enums_entities) == number_of_defined_enum_values
+    assert len(undefined) == number_of_undefined_default_enum_values
+
+
+def test_issue_293():
```

Comment:
Test name should shortly reflect what behaviour is tested, with docstring (or plain comment) mentioning that its a regression test for issue, eg.:

```
def test_something():
    # regression test for https://github.com/mirumee/issues/31321
```

Reply:
Done.